Honey4Groceries/Sources/Services/LocationService.swift

```diff
@@ -0,0 +1,18 @@
+import PromiseKit
+
+class LocationService {
+    func getLocation() -> Guarantee<CLLocation?> {
```

Comment:
don't return nil, create an enum of errors and throw instead. for your reference

enum VideoError: Error {
    case invalidStart
    case invalidStop
    case nilInput
    case nilOutput
    case accessDenied
}